lightning/src/offers/invoice.rs

```diff
@@ -469,8 +470,14 @@ impl Invoice {
 		self.signature
 	}
 
+	/// Verifies that the invoice was for a request or refund created using the given key.
+	#[allow(unused)]
+	pub(crate) fn verify(&self, key: &ExpandedKey) -> bool {
```

Comment:
In the upcoming PR, I have `NodeSigner::sign_bolt12_message` given the metadata, so it could extract it from there. Although, soon it will need to be decrypted first unless the signer could do that.

Otherwise, if returned here, we'd have callers pipe it into `UnsignedInvoiceRequest::sign`. Though, with the recent change for deriving the keys, I was considering leveraging the `Option<KeyPair>` field that I added to `UnsignedInvoiceRequest` instead of exposing the `Nonce`.